test_integration/lib/test/realtime/realtime_auth_url_test.dart

```diff
@@ -0,0 +1,42 @@
+
+import 'dart:io';
+
+import 'package:ably_flutter/ably_flutter.dart';
+import 'package:ably_flutter_integration_test/app_provisioning.dart';
+import 'package:ably_flutter_integration_test/factory/reporter.dart';
+
+//Make sure that creating realtime with authURL won't crash
+Future<Map<String, dynamic>> testCreateRealtimeWithAuthUrl({
+  required Reporter reporter,
+  Map<String, dynamic>? payload,
+}) async {
+  reporter.reportLog('init start');
+  final appKey = await AppProvisioning().provisionApp();
+
+  final clientOptionsForToken = ClientOptions(
+    key: appKey,
+    environment: 'sandbox',
+    logLevel: LogLevel.verbose,
+  );
+
+  final ablyForToken = Rest(
+    options: clientOptionsForToken,
+  );
+  final accessToken = await ablyForToken.auth.requestToken();
+
+  final Map<String, String> headers = {
+    HttpHeaders.authorizationHeader: "Bearer $accessToken",
+    HttpHeaders.contentTypeHeader: "application/json",
+    HttpHeaders.acceptLanguageHeader: "ios",
+  };
+  final options = ClientOptions(
+    authUrl: 'https://auth.ably.io/auth',
+    authHeaders: headers,
+    echoMessages: false,
+  );
+  final realtime = Realtime(options: options);
+  realtime.connection.on().listen((stateChange) {
```

Comment:
Does this test actually wait for the client to be connected? It looks like you're just adding a listener to log `stateChange`s to the console but not actually checking whether the client becomes connected

Reply:
Not really. This test is only verifying that app won't crash in case of an authUrl is passed as described https://github.com/ably/ably-flutter/pull/466/commits/746fef4aab5901ff126f048ec7dc56e1358cb048

Reply:
Tests for connections were added